File: Champions/Gangplank/Q.lua
Vector2 = require 'Vector2' -- include 2d vector lib 

function onStartCasting()   
    
end

function onFinishCasting()
    local target = castTarget
    addProjectileTarget("pirate_parley_mis", target, false) 	
end

function applyEffects()
    local target = castTarget
    local damage = owner:GetStats().AttackDamage.Total + -5 + (25*spellLevel)
    local newGold = owner:GetStats().Gold + 3 + (1*spellLevel)
	
    if ((not (target == 0)) and (not isDead(target))) then
        if castTarget:GetStats().CurrentHealth > damage then
            owner:dealDamageTo(target, damage, TYPE_PHYSICAL, SOURCE_ATTACK, false)
        else
            owner:GetStats().Gold = newGold
            owner:dealDamageTo(target, damage, TYPE_PHYSICAL, SOURCE_ATTACK, false)
        end    
    end

    destroyProjectile()
end

function onUpdate(diff)

end

File: Champions/Gangplank/Q.cs
using System;
using GameServerCore;
using GameServerCore.Enums;
using LeagueSandbox.GameServer.GameObjects.AttackableUnits;
using LeagueSandbox.GameServer.GameObjects.AttackableUnits.AI;
using LeagueSandbox.GameServer.GameObjects.Missiles;
using LeagueSandbox.GameServer.GameObjects.Spells;
using LeagueSandbox.GameServer.Scripting.CSharp;

namespace Spells
{
    public class Parley : IGameScript
    {
        public void OnActivate(Champion owner)
        {
        }

        public void OnDeactivate(Champion owner)
        {
        }

        public void OnStartCasting(Champion owner, Spell spell, AttackableUnit target)
        {
        }

        public void OnFinishCasting(Champion owner, Spell spell, AttackableUnit target)
        {
            spell.AddProjectileTarget("pirate_parley_mis", target);
        }

        public void ApplyEffects(Champion owner, AttackableUnit target, Spell spell, Projectile projectile)
        {
            var isCrit = new Random().Next(0, 100) < (owner.Stats.CriticalChance.Total * 100);
            var baseDamage = new[] {20, 45, 70, 95, 120}[spell.Level - 1] + owner.Stats.AttackDamage.Total;
            var damage = new Damage(isCrit ? baseDamage * owner.Stats.CriticalDamage.Total / 100 : baseDamage,
                DamageType.DAMAGE_TYPE_PHYSICAL, DamageSource.DAMAGE_SOURCE_ATTACK, isCrit);
            var goldIncome = new[] {4, 5, 6, 7, 8}[spell.Level - 1];
            if (target != null && !target.IsDead)
            {
                target.TakeDamage(owner, damage);
                if (target.IsDead)
                {
                    owner.Stats.Gold += goldIncome;
                    var manaCost = new float[] {50, 55, 60, 65, 70}[spell.Level - 1];
                    var newMana = owner.Stats.CurrentMana + manaCost / 2;
                    var maxMana = owner.Stats.ManaPoints.Total;
                    if (newMana >= maxMana)
                    {
                        owner.Stats.CurrentMana = maxMana;
                    }
                    else
                    {
                        owner.Stats.CurrentMana = newMana;
                    }
                }

                projectile.SetToRemove();
            }
        }

        public void OnUpdate(double diff)
        {
        }
    }
}
